src/main/java/org/jboss/pnc/api/reqour/dto/RepositoryCloneResponseCallback.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2024-2024 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.api.reqour.dto;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.Builder;
+import lombok.Value;
+import lombok.extern.jackson.Jacksonized;
+import org.jboss.pnc.api.reqour.dto.validation.ExternalURL;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
+/**
+ * DTO used as a response of the reqour's repository clone operation. It's sent as a callback to the invoker.
+ */
+@Builder(toBuilder = true)
+@Value
+@Jacksonized
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class RepositoryCloneResponseCallback {
+
+    /**
+     * Type of scm: usually: git
+     */
+    @NotBlank
+    String type;
+
+    /**
+     * Original repository to sync from
+     */
+    @ExternalURL
+    String originRepoUrl;
+
+    /**
+     * Repository to sync to
+     */
+    @NotBlank
```

Comment:
Shouldn't this be `@ExternalURL`? (in particular, it should match SCP-like format, shouldn't it?)

Reply:
Right! but then the ExternalUrl name might cause confusion since `targetRepoUrl` is an internal url

Reply:
I've renamed `@ExternalURL` to `@ValidGitRepositoryURL` (as we've agreed in Slack DMs). Also, I've added this validation check where appropriate - see [a690754](https://github.com/project-ncl/pnc-api/pull/231/commits/a690754b328a2415f0afebd426dea80cb83f9fb7).

Hence, merging the PR